contracts/minters/base-minter/src/contract.rs

```diff
@@ -107,61 +111,146 @@ pub fn execute(
     msg: ExecuteMsg,
 ) -> Result<Response, ContractError> {
     match msg {
-        ExecuteMsg::Mint { token_uri } => execute_mint_sender(deps, info, token_uri),
+        ExecuteMsg::Mint { token_uri } => execute_mint_sender(deps, env, info, token_uri),
         ExecuteMsg::UpdateStartTradingTime(time) => {
             execute_update_start_trading_time(deps, env, info, time)
         }
+        ExecuteMsg::ReceiveNft(msg) => execute_burn_to_mint(deps, env, info, msg),
     }
 }
 
+pub fn execute_burn_to_mint(
+    _deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: Cw721ReceiveMsg,
+) -> Result<Response, ContractError> {
+    let mut res = Response::new();
+    let burn_msg = cw721::Cw721ExecuteMsg::Burn {
+        token_id: msg.token_id,
+    };
+    let cosmos_burn_msg = CosmosMsg::Wasm(WasmMsg::Execute {
+        contract_addr: info.sender.to_string(),
+        msg: to_binary(&burn_msg)?,
+        funds: vec![],
+    });
+    res = res.add_message(cosmos_burn_msg);
+
+    let token_uri_msg: TokenUriMsg = from_binary(&msg.msg)?;
+    let execute_mint_msg = ExecuteMsg::Mint {
+        token_uri: token_uri_msg.token_uri,
+    };
+    let cosmos_mint_msg = CosmosMsg::Wasm(WasmMsg::Execute {
+        contract_addr: env.contract.address.to_string(),
+        msg: to_binary(&execute_mint_msg)?,
+        funds: vec![],
+    });
+    let res = res.add_message(cosmos_mint_msg);
+    Ok(res)
+}
+
 pub fn execute_mint_sender(
     deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    token_uri: String,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    if matches!(config.mint_price, Token::NonFungible(_)) {
```

Comment:
Nope, mint_price is a token, token looks like this: 
```rs
pub enum Token {
    Fungible(Coin),
    NonFungible(String),
}
```
So if you get a NonFungible token it will fail, but Fungible token won't match the pattern and therefore this check will pass.

Reply:
This redundant check was removed per Discord convo